azure-iot-device/samples/async-pnp-samples/sample.py

```diff
@@ -193,33 +191,35 @@ def __init__(self, interface_instance_name):
 
     listeners = asyncio.ensure_future(
         asyncio.gather(
-            environmental_command_listener(environmental_sensor),
-            environmental_property_changed_listener(environmental_sensor),
-            model_definition_command_listener(model_definition),
+            environmental_model_listener(environmental_sensor),
+            model_definition_listener(model_definition),
         )
     )
 
-    await environmental_sensor.state.report(True)
-    await device_information.manufacturer.report("Contoso Device Corporation")
-    await device_information.model.report("Contoso 4762B-turbo")
-    await device_information.sw_version.report("3.1")
-    await device_information.os_name.report("ContosoOS")
-    await device_information.processor_architecture.report("4762")
-    await device_information.processor_manufacturer.report("Contoso Foundries")
-    await device_information.total_storage.report("64000")
-    await device_information.total_memory.report("640")
+    environmental_sensor.properties.state = True
+    await environmental_sensor.report_properties()  # only reports changed properties
+
+    device_information.properties.manufacturer = "Contoso Device Corporation"
+    device_information.properties.model = "Contoso 4762B-turbo"
+    device_information.properties.sw_version = "3.1"
+    device_information.properties.os_name = "ContosoOS"
+    device_information.properties.processor_architecture = "4762"
+    device_information.properties.processor_manufacturer = "Contoso Foundries"
+    device_information.properties.total_storage = "64000"
+    device_information.properties.total_memory = "640"
+    await device_information.report_properties()  # only repors changed properties
 
     #  send telemetry every 5 seconds
     def send_telemetry():
         while True:
-            await environmental_sensor.sendTelemetry(
-                {"temp": 10 + random.random_int(0, 90), "humid": 1 + random.randint(0, 99)}
-            )
+            environmental_sensor.telemetry.temp = 10 + random.random_int(0, 90)
+            environmental_sensor.telemetry.humid = 1 + random.randint(0, 99)
+            await environmental_sensor.report_telemetry()  # only sends telemetry values that have changed
```

Comment:
>await environmental_sensor.report_telemetry()  # only sends telemetry values that have changed [](start = 12, length = 94)

Don't forget the "force" flag we discussed to update everything regardless if it changed or not